compiler/src/iree/compiler/Codegen/Common/GPUDistributeSharedMemoryCopy.cpp

```diff
@@ -39,15 +39,21 @@ static void populateTilingCopyToWorkgroupMemPatterns(
   // Tile and distribute copy to workgroup memory.
   linalg::TileSizeComputationFunction wgCopyTileSizeFn =
       [](OpBuilder &builder, Operation *operation) {
-        const int64_t copyTileSize = 4;
+        int64_t copyTileSize;
         // We tile to 4 as we want each thread to load 4 element in a cyclic
         // distribution.
         SmallVector<Value, 4> tileSizesVal;
-        unsigned rank = cast<linalg::GenericOp>(operation)
-                            .getOperand(0)
-                            .getType()
-                            .cast<MemRefType>()
-                            .getRank();
+        MemRefType lhsMemRefType = cast<linalg::GenericOp>(operation)
```

Comment:
Is it safe to say you are trying a 128 bit copy. So maybe just do 128 / (element type bitwidth) here?

Reply:
Makes sense, I am just unsure, how to accomplish the 128 bit copy with the existing functions

Reply:
+1 to what Mahesh said, you can use `getElementTypeBitWidth` to get the bit size and deduce the number of elements based on that. And we can have a common function to get the value across the different cases.